app/src/main/java/com/example/dungeontest/Settings.kt

```diff
@@ -78,34 +131,83 @@ fun SettingsScreen(drawerState: DrawerState, scope: CoroutineScope) {
                 }
             )
-        }
+        },
+        floatingActionButton = {
+            ExtendedFloatingActionButton(onClick = {
+                Log.d("SettingsScreen", "tokenValue: ${tokenValue.value.text}")
+                Log.d("SettingsScreen", "selectedCard: $selectedCard")
+                CoroutineScope(Dispatchers.IO).launch {
+                    preferences.saveToken(tokenValue.value.text)
+                    preferences.saveSelectedModel(selectedCard.value)
+                    withContext(Dispatchers.Main) {
+                        if (snackbarHostState.currentSnackbarData == null) {
+
+                            snackbarHostState.showSnackbar("Data saved successfully")
+                        }
+                    }
+                }
+            }) {
+                Icon(Icons.Filled.Check, contentDescription = "Save")
+                Text("Save")
+
+            }
+        },
+        floatingActionButtonPosition = FabPosition.Center
     ) { innerPadding ->
         Box(
             modifier = Modifier
                 .fillMaxSize()
                 .padding(innerPadding)
+
         ) {
             Column(
                 verticalArrangement = Arrangement.spacedBy(16.dp),
                 horizontalAlignment = Alignment.CenterHorizontally,
                 modifier = Modifier.fillMaxSize()
             ) {
-                Spacer(modifier = Modifier.height(2.dp))
-                SimpleOutlinedTextFieldSample()
+                Spacer(modifier = Modifier.width(2.dp))
+                SimpleOutlinedTextFieldSample(tokenValue)
+                Spacer(modifier = Modifier.width(2.dp))
+                if (isLandscape) {
+                    LazyRow(
+                        horizontalArrangement = Arrangement.Center,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(start = 20.dp, end = 20.dp)
+
+                    ) {
+                        items(cardInfos) { models ->
+                            OutlinedCardExample(models.id, models.title, models.description, selectedCard)
+                            Spacer(modifier = Modifier.width(20.dp))
+                        }
+                    }
+                } else {
+                    LazyColumn(
+                        verticalArrangement = Arrangement.spacedBy(16.dp),
+                        horizontalAlignment = Alignment.CenterHorizontally,
+                        modifier = Modifier.fillMaxSize()
+                    ) {
+                        items(cardInfos) { models ->
+                            OutlinedCardExample(models.id, models.title, models.description, selectedCard)
+                        }
+                    }
+                }
             }
+
         }
     }
 }
 
 @Composable
-fun SimpleOutlinedTextFieldSample() {
-    var text by remember { mutableStateOf("") }
+fun SimpleOutlinedTextFieldSample(tokenValue: MutableState<TextFieldValue>) {
```

Comment:
b4 merge, could you please make the composable name less generic and more clear about its purpose?